src/SignalR/clients/csharp/Client/test/UnitTests/TimerAwaitableTests.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Internal;
+using Xunit;
+
+namespace Microsoft.AspNetCore.SignalR.Client.Tests
+{
+    public class TimerAwaitableTests
+    {
+        [Fact]
+        public void FinalizerRunsIfTimerAwaitableReferencesObject()
+        {
+            var tcs = new TaskCompletionSource<object>(TaskCreationOptions.RunContinuationsAsynchronously);
+            UseTimerAwaitableAndUnref(tcs);
+
+            // Make sure it *really* cleans up
+            for (int i = 0; i < 5; i++)
+            {
+                GC.Collect();
+                GC.WaitForPendingFinalizers();
```

Comment:
```suggestion
                GC.Collect();
                GC.WaitForPendingFinalizers();
                if (tcs.Task.IsCompleted)
                {
                	return;
                }
```

That way we can exit early if the task finishes early.